src/main/java/com/lothrazar/cyclic/block/cable/item/TileCableItem.java

```diff
@@ -53,44 +51,39 @@ private static ItemStackHandler createHandler() {
     return new ItemStackHandler(1);
   }
 
-  List<Integer> rawList = IntStream.rangeClosed(
-      0,
-      5).boxed().collect(Collectors.toList());
-
   @Override
   public void tick() {
-    for (Direction extractSide : Direction.values()) {
-      EnumConnectType connection = this.getBlockState().get(CableBase.FACING_TO_PROPERTY_MAP.get(extractSide));
+    for (final Direction extractSide : Direction.values()) {
+      final EnumProperty<EnumConnectType> extractFace = CableBase.FACING_TO_PROPERTY_MAP.get(extractSide);
+      final EnumConnectType connection = this.getBlockState().get(extractFace);
       if (connection.isExtraction()) {
-        IItemHandler sideHandler = flow.get(extractSide).orElse(null);
+        final IItemHandler sideHandler = flow.get(extractSide).orElse(null);
         tryExtract(sideHandler, extractSide, extractQty, filter);
       }
     }
     normalFlow();
   }
 
   private void normalFlow() {
-    IItemHandler sideHandler;
-    Direction outgoingSide;
-    for (Direction incomingSide : Direction.values()) {
-      sideHandler = flow.get(incomingSide).orElse(null);
-      //thise items came from that
-      Collections.shuffle(rawList);
-      boolean validAdjacent = false;
-      for (Integer i : rawList) {
-        outgoingSide = Direction.values()[i];
-        if (outgoingSide == incomingSide) {
+    incomingSideLoop: for (final Direction incomingSide : Direction.values()) {
+      //in all cases sideHandler is required
+      final IItemHandler sideHandler = flow.get(incomingSide).orElse(null);
+
+      for (final Direction outgoingSide : UtilDirection.inDifferingOrder.next()) {
+        if (outgoingSide == incomingSide)
           continue;
-        }
-        EnumConnectType connection = this.getBlockState().get(CableBase.FACING_TO_PROPERTY_MAP.get(outgoingSide));
-        if (connection.isExtraction() || connection.isBlocked()) {
+
+        final EnumProperty<EnumConnectType> outgoingFace = CableBase.FACING_TO_PROPERTY_MAP.get(outgoingSide);
+        final EnumConnectType outgoingConnection = this.getBlockState().get(outgoingFace);
+        if (outgoingConnection.isExtraction() || outgoingConnection.isBlocked())
           continue;
-        }
-        validAdjacent = validAdjacent || this.moveItems(outgoingSide, 64, sideHandler);
```

Comment:
Once validAdjacent has been set to true, it keeps looping without moving any more items.

Reply:
Also 64 has been hardcoded, so swapped it out for FLOW_QTY.